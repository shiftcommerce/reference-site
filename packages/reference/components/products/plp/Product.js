// Libraries
import { Component } from 'react'
import { Highlight } from 'react-instantsearch/dom'
import classNames from 'classnames'
import Link from 'next/link'

// Objects
import Image from '../../../objects/Image'
import Button from '../../../objects/Button'

class Product extends Component {
  renderAssetFile (hit) {
    let asset = hit.product_assets[0]

    return (
      <div className='c-product__image-container'>
        <Image
          className={'c-product__image'}
          src={(asset && asset.url) || '/static/placeholder.png'}
          alt={(asset && asset.alt_text)}
          aria-label={(asset && asset.alt_text)}
        />
      </div>
    )
  }

  renderProductPrice (hit) {
    let metaData = hit.variant_meta_data
    let price = (((metaData && metaData.eu && metaData.eu.price) || 0) / 100).toFixed(2)

    return (
      <div className='c-product__price'>
        &pound;{ price }
      </div>
    )
  }

  renderRatingStars (hit) {
    let rating = (hit.product_rating || 0)

    return (
      [1, 2, 3, 4, 5].map((key, index) =>
        <span key={key} className={classNames({'c-product__rating--fill': key <= rating, 'c-product__rating--empty': key > rating})}>
          &#9733;
        </span>
      )
    )
  }

  render () {
    let {
      hit
    } = this.props

    let productID = hit.objectID.split('_')[0]

    return (
      <div className='c-product'>
        <Link href={`/products/product?id=${productID}`} as={hit.product_path}>
          <a>
            { this.renderAssetFile(hit) }

            <div className='c-product__title'>
              <Highlight attributeName='product_title' hit={hit} />
            </div>

            <div className='c-product__sku'>
              <Highlight attributeName='product_meta_data.eu.sku' hit={hit} />
            </div>

            { this.renderProductPrice(hit) }

            <div className='c-product__view-details'>
              <Button label='View' />
            </div>

            { this.renderRatingStars(hit) }
          </a>
        </Link>
      </div>
    )
  }
}

export default Product
