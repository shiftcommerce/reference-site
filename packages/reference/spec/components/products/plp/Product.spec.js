// Component
import Product from '../../../../components/products/plp/Product'

// Fixtures
import productSearchHit from '../../../fixtures/productSearchHit.fixture'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = shallow(
    <Product hit={productSearchHit} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
})
