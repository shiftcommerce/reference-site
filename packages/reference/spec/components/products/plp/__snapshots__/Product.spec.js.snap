// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="c-product"
>
  <Link
    as="/products/a2d8d304-19eb-4a98-ae77-d7db53715cc4"
    href="/products/product?id=a2d8d304-19eb-4a98-ae77-d7db53715cc4"
  >
    <a>
      <div
        className="c-product__image-container"
      >
        <Image
          alt="test alt value"
          aria-label="test alt value"
          className="c-product__image"
          src="https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12671/S2623408_C29P_Alt2.jpg"
        />
      </div>
      <div
        className="c-product__title"
      >
        <AlgoliaHighlighter(Highlight)
          attributeName="product_title"
          hit={
            Object {
              "_highlightResult": Object {
                "product_meta_data": Object {
                  "eu": Object {
                    "sku": Object {
                      "matchLevel": "none",
                      "matchedWords": Array [],
                      "value": "SK8407636324",
                    },
                  },
                },
                "product_title": Object {
                  "fullyHighlighted": false,
                  "matchLevel": "full",
                  "matchedWords": Array [
                    "long",
                  ],
                  "value": "<ais-highlight-0000000000>Long</ais-highlight-0000000000> Shirt",
                },
                "variant_sku": Object {
                  "matchLevel": "none",
                  "matchedWords": Array [],
                  "value": "SK8407636324-2",
                },
              },
              "category_ids": Array [
                "22bca338-b578-4aee-a302-992764fa3e72",
              ],
              "objectID": "a2d8d304-19eb-4a98-ae77-d7db53715cc4_6d2a54b2-3988-4e3c-b4d8-26feb6ea852e",
              "product_assets": Array [
                Object {
                  "alt_text": "test alt value",
                  "id": 1,
                  "url": "https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12671/S2623408_C29P_Alt2.jpg",
                },
              ],
              "product_meta_data": Object {
                "eu": Object {
                  "colour": "Blue",
                  "description": "Combining comfort and practicality, this grey marl textured t-shirt is a go-to everyday essential. Featuring a v-neck and front pocket, this staple piece can be layered to create a range different looks.",
                  "sku": "SKU2659078",
                },
              },
              "product_reference": "product_reference",
              "product_title": "Textured Long T-Shirt",
              "variant_assets": Array [],
              "variant_meta_data": Object {
                "eu": Object {
                  "price": 10,
                  "size": "Size 8",
                },
              },
              "variant_sku": "small_1",
              "variant_title": "Small",
            }
          }
        />
      </div>
      <div
        className="c-product__sku"
      >
        <AlgoliaHighlighter(Highlight)
          attributeName="product_meta_data.eu.sku"
          hit={
            Object {
              "_highlightResult": Object {
                "product_meta_data": Object {
                  "eu": Object {
                    "sku": Object {
                      "matchLevel": "none",
                      "matchedWords": Array [],
                      "value": "SK8407636324",
                    },
                  },
                },
                "product_title": Object {
                  "fullyHighlighted": false,
                  "matchLevel": "full",
                  "matchedWords": Array [
                    "long",
                  ],
                  "value": "<ais-highlight-0000000000>Long</ais-highlight-0000000000> Shirt",
                },
                "variant_sku": Object {
                  "matchLevel": "none",
                  "matchedWords": Array [],
                  "value": "SK8407636324-2",
                },
              },
              "category_ids": Array [
                "22bca338-b578-4aee-a302-992764fa3e72",
              ],
              "objectID": "a2d8d304-19eb-4a98-ae77-d7db53715cc4_6d2a54b2-3988-4e3c-b4d8-26feb6ea852e",
              "product_assets": Array [
                Object {
                  "alt_text": "test alt value",
                  "id": 1,
                  "url": "https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12671/S2623408_C29P_Alt2.jpg",
                },
              ],
              "product_meta_data": Object {
                "eu": Object {
                  "colour": "Blue",
                  "description": "Combining comfort and practicality, this grey marl textured t-shirt is a go-to everyday essential. Featuring a v-neck and front pocket, this staple piece can be layered to create a range different looks.",
                  "sku": "SKU2659078",
                },
              },
              "product_reference": "product_reference",
              "product_title": "Textured Long T-Shirt",
              "variant_assets": Array [],
              "variant_meta_data": Object {
                "eu": Object {
                  "price": 10,
                  "size": "Size 8",
                },
              },
              "variant_sku": "small_1",
              "variant_title": "Small",
            }
          }
        />
      </div>
      <div
        className="c-product__price"
      >
        £
        10.00
      </div>
      <div
        className="c-product__view-details"
      >
        <Button
          label="View"
        />
      </div>
      <span
        className="c-product__rating--empty"
      >
        ★
      </span>
      <span
        className="c-product__rating--empty"
      >
        ★
      </span>
      <span
        className="c-product__rating--empty"
      >
        ★
      </span>
      <span
        className="c-product__rating--empty"
      >
        ★
      </span>
      <span
        className="c-product__rating--empty"
      >
        ★
      </span>
    </a>
  </Link>
</div>
`;
